[CI] HttOptionsNoAuthnIntegTests testNoAuthnForPreFlightRequest failing

**Build Scans:**
- [elasticsearch-intake #38694 / part4](https://gradle-enterprise.elastic.co/s/rrcqafexovfpk)
- [elasticsearch-pull-request #140624 / part-4](https://gradle-enterprise.elastic.co/s/it2efb2u4fgam)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:security:internalClusterTest" --tests "org.elasticsearch.xpack.security.authc.HttOptionsNoAuthnIntegTests.testNoAuthnForPreFlightRequest" -Dtests.seed=711EFDA28739AB37 -Dtests.locale=xh-ZA -Dtests.timezone=Africa/Ouagadougou -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.security.authc.HttOptionsNoAuthnIntegTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testNoAuthnForPreFlightRequest'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.IllegalStateException: org.elasticsearch.discovery.MasterNotDiscoveredException
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testNoAuthnForPreFlightRequest
- [main] 1 consecutive failures in step part4
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testNoAuthnForPreFlightRequest (0.2% fail rate in 839 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -2147483648 out of bounds for length 32	
	at java.util.Date.toString(Date.java:1036) ~[?:?]	
	at sun.security.x509.CertificateValidity.valid(CertificateValidity.java:182) ~[?:?]	
	at sun.security.x509.X509CertImpl.checkValidity(X509CertImpl.java:534) ~[?:?]	
	at sun.security.ssl.X509KeyManagerCertChecking$CheckType.check(X509KeyManagerCertChecking.java:380) ~[?:?]	
	at sun.security.ssl.X509KeyManagerCertChecking.certificateCheck(X509KeyManagerCertChecking.java:255) ~[?:?]	
	at sun.security.ssl.X509KeyManagerCertChecking.checkAlias(X509KeyManagerCertChecking.java:162) ~[?:?]
```

Should be addressed by https://github.com/elastic/elasticsearch/pull/147819

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -523,12 +523,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.commits.StatelessCommitServiceTests
   method: testOldCommitsAreRetainedIfSearchNodesUseThem
   issue: https://github.com/elastic/elasticsearch/issues/147888
-- class: org.elasticsearch.xpack.security.authc.HttOptionsNoAuthnIntegTests
-  method: testNoAuthnForPreFlightRequest
-  issue: https://github.com/elastic/elasticsearch/issues/147898
-- class: org.elasticsearch.xpack.security.authc.HttOptionsNoAuthnIntegTests
-  method: testNoAuthnForResourceOptionsMethod
-  issue: https://github.com/elastic/elasticsearch/issues/147899
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   method: test {yaml=synonyms/110_synonyms_invalid/Load index with an invalid synonym rule with lenient set to false}
   issue: https://github.com/elastic/elasticsearch/issues/147907

